lightning-background-processor/src/lib.rs

```diff
@@ -791,6 +835,69 @@ mod tests {
 		assert!(bg_processor.stop().is_ok());
 	}
 
+	#[test]
+	fn test_not_pruning_network_graph_until_graph_sync_completion() {
+		let nodes = create_nodes(2, "test_not_pruning_network_graph_until_graph_sync_completion".to_string());
+		let data_dir = nodes[0].persister.get_data_dir();
+		let (sender, receiver) = std::sync::mpsc::sync_channel(1);
+		let persister = Arc::new(Persister::new(data_dir.clone()).with_graph_persistence_notifier(sender));
+		let network_graph = nodes[0].network_graph.clone();
+		let rapid_sync = Arc::new(RapidGossipSync::new(network_graph.clone()));
+		let features = ChannelFeatures::empty();
+		network_graph.add_channel_from_partial_announcement(42, 53, features, nodes[0].node.get_our_node_id(), nodes[1].node.get_our_node_id())
+			.expect("Failed to update channel from partial announcement");
+		let original_graph_description = network_graph.to_string();
+		assert!(original_graph_description.contains("42: features: 0000, node_one:"));
```

Comment:
This assertion doesn't seem necessary.

Reply:
It helps assert that channel 42 is actually there.